Examples/Io/Csv/include/ActsExamples/Io/Csv/CsvSimHitWriter.hpp

```diff
@@ -0,0 +1,64 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "ActsExamples/EventData/GeometryContainers.hpp"
+#include "ActsExamples/EventData/SimHit.hpp"
+#include "ActsExamples/Framework/WriterT.hpp"
+
+#include <limits>
+#include <string>
+
+namespace ActsExamples {
+
+/// Write out a simhit collection before detector digitization in comma-
+/// separated-value format.
+///
+/// This writes one file per event containing information about the
+/// global space points, momenta (before and after interaction) and hit index
+/// into the configured output directory. By default it writes to the
+/// current working directory. Files are named using the following schema
+///
+///     event000000001-<stem>.csv
+///     event000000002-<stem>.csv
+///     ...
+///
+/// and each line in the file corresponds to one simhit.
+class CsvSimHitWriter final : public WriterT<SimHitContainer> {
+ public:
+  struct Config {
+    /// Which simulated (truth) hits collection to use.
+    std::string inputSimulatedHits;
```

Comment:
```suggestion
    std::string inputSimHits;
```
For consistency with other algorithms.